### PR TITLE
Client documentation updates

### PR DESCRIPTION
**Pull request by jeffreylovitz**
No rush on this!

Clients have been elevated to a dedicated link, and I added a technical page with details for writing a client implementation (long overdue).

### CODE REVIEW

File: docs/client_spec.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+Additionally, two enums are exposed:
+
+[ColumnType](https://github.com/RedisGraph/RedisGraph/blob/ff108d7e21061025166a35d29be1a1cb5bac6d55/src/resultset/formatters/resultset_formatter.h#L14-L19) indicates what type of value is held in each column (more formally, that offset into each row of the result set). Each entry in the header row will be a 2-array, with this enum in the first position and the column name string in the second.
+
+[PropertyType](https://github.com/RedisGraph/RedisGraph/blob/ff108d7e21061025166a35d29be1a1cb5bac6d55/src/resultset/formatters/resultset_formatter.h#L21-L28) indicates the data type (such as integer or string) of each returned scalar value. Each scalar values is emitted as a 2-array, with this enum in the first position and the actual value in th second. A column can consist exclusively of scalar values, such as both of the columns created by `RETURN a.value, 'this literal string'`. Each property on a graph entity also has a scalar as its value, so this construction is nested in each value of the properties array when a column contains a node or relationship.
```
Comment (K-Jo) [inline docs/client_spec.md]: `th second`  -> the

File: docs/client_spec.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+As such, the complete representation is as follows:
+
+1. The relation's internal ID.
+2. The relationship type ID.
+3. The source node's internal ID.
+4. The destination node's internal ID.
+5. The key-value pairs of all properties the relation possesses.
+
+```sh
+[
```
Comment (K-Jo) [inline docs/client_spec.md]: don't think the indentations are correct here

File: docs/client_spec.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+6. "Relationships created: (integer)"
+7. "Query internal execution time: (float) milliseconds"
+
+## Procedure Calls
+
+Property keys, node labels, and relationship types are all returned as IDs rather than strings in the compact format. For each of these 3 string-ID mappings, IDs start at 0 and increase monotonically.
+
+As such, the client should store an string array for each of these 3 mappings, and print the appropriate string for the user by checking an array at position _ID_. If an ID greater than the array length is encountered, the local array should be updated with a procedure call.
+
+These calls are described generally in the [Procedures documentaiton](commands.md#procedures).
```
Comment (K-Jo) [inline docs/client_spec.md]: documentaiton

File: docs/client_spec.md (hunk 1 above)
Comment (jeffreylovitz) [reply docs/client_spec.md]: Ugh, good catches! Sometimes I miss text editors that have spell check ;)

**Pull request merged**